src/cpp/CMakeLists.txt

```diff
@@ -285,6 +285,12 @@ elseif(NOT EPROSIMA_INSTALLER)
         $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
         )
 
+    # Define option to link pthread library when sanitizers are enabled
+    option(SANITIZER_ENABLED "Add pthread library when sanitizers are enabled" OFF)
+    if(SANITIZER_ENABLED)
+        target_link_libraries(${PROJECT_NAME} ${PRIVACY} pthread)
```

Comment:
Can you use a cmake generator expression to have only one `target_link_libraries` call? As we do in line 284. Thanks

Reply:
Done